nomad/structs/csi.go

```diff
@@ -652,17 +652,50 @@ func (p *CSIPlugin) AddPlugin(nodeID string, info *CSIInfo) {
 // DeleteNode removes all plugins from the node. Called from state.DeleteNode in a
 // transaction
 func (p *CSIPlugin) DeleteNode(nodeID string) {
+	p.DeleteNodeType(nodeID, CSIPluginTypeMonolith)
+}
+
+// DeleteNodeType deletes all plugins of type from the node. Called from deleteJobFromPlugin
```

Comment:
This code makes sense but the unfortunate name collision between Nomad client node and CSI node plugins makes the name and docstring hard to understand. What do you think of this? 

```go
// DeleteNodeForType deletes a client node from the list of controllers or node instance of a plugin.
// Called from deleteJobFromPlugin during job deregistration.
func (p *CSIPlugin) DeleteNodeForType(nodeID string, pluginType CSIPluginType) {
```

Reply:
wfm